### DISCUSSION TITLE
How to revoke user's tokens?

### DISCUSSION THREAD

**Discussion by khankamil** [aboutyou/dart_packages#319] [Q&A]
Hi

I have been using this package in my iOS App. But I want to add a feature to let user delete their account. For this I need to let user initiate to revoke the token. Any idea how I can do this?

Thanks.

**Comment by dalriada33** [comment]
> Hi
> 
> I have been using this package in my iOS App. But I want to add a feature to let user delete their account. For this I need to let user initiate to revoke the token. Any idea how I can do this?
> 
> Thanks.

I would like to do this too. Have you found a solution for that?

**Comment by barahijawi** [comment]
the user email and name and other information returned by AppleLogin will be null after the first time if we didn't revoke the access_token for that user.

all you need to do, is to revoke the access token which has been generated at the first login and then try to login once again.

you may use the following request for **getting that access token** :

     await post(
        'https://appleid.apple.com/auth/token',
        {
          'client_id': com/package.name,
          'client_secret': clientSecret,
          'code': authorizationCode,
          'grant_type': 'authorization_code',
          'redirect_uri': ''
        },

and use this request for **revoking the access token**:

     'https://appleid.apple.com/auth/revoke'
    'client_id': com.package.name,
          'client_secret': clientSecret,
          'token':token



you will need to generate client secret using private key, you may follow this link for how generate that. token:
[https://developer.apple.com/documentation/accountorganizationaldatasharing/creating-a-client-secret][1]


this also helpful : [Medium article on how to generate client secret through dart][2]


  [1]: https://developer.apple.com/documentation/accountorganizationaldatasharing/creating-a-client-secret
  [2]: https://martin-robert-fink.medium.com/generating-apple-client-secret-with-dart-28f29a55f8f0
